src/Microsoft.ML.Data/Data/Conversion.cs

```diff
@@ -1472,61 +1454,37 @@ public bool TryParse(ref TX src, out R8 dst)
 
         public bool TryParse(ref TX src, out TS dst)
         {
+            dst = default;
             if (!src.HasChars)
-            {
-                if (src.IsNA)
-                    dst = TS.NA;
-                else
-                    dst = default(TS);
                 return true;
-            }
-            TimeSpan res;
-            if (TimeSpan.TryParse(src.ToString(), CultureInfo.InvariantCulture, out res))
-            {
-                dst = new TS(res);
+
+            if (TimeSpan.TryParse(src.ToString(), CultureInfo.InvariantCulture, out dst))
                 return true;
-            }
-            dst = TS.NA;
+
             return IsStdMissing(ref src);
         }
 
         public bool TryParse(ref TX src, out DT dst)
         {
+            dst = default;
             if (!src.HasChars)
-            {
-                if (src.IsNA)
-                    dst = DvDateTime.NA;
-                else
-                    dst = default(DvDateTime);
                 return true;
-            }
-            DateTime res;
-            if (DateTime.TryParse(src.ToString(), CultureInfo.InvariantCulture, DateTimeStyles.AssumeUniversal | DateTimeStyles.AdjustToUniversal, out res))
-            {
-                dst = new DT(res);
+
+            if (DateTime.TryParse(src.ToString(), CultureInfo.InvariantCulture, DateTimeStyles.AssumeUniversal | DateTimeStyles.AdjustToUniversal, out dst))
                 return true;
-            }
-            dst = DvDateTime.NA;
+
             return IsStdMissing(ref src);
```

Comment:
>IsStdMissing(ref src); [](start = 19, length = 22)

So I feel like if you parse missing into a value type that does not support missing, we might expect the result to be that we would throw, rather than just the default value.

Reply:
I didn't really care too much here because of the DvText change that would remove missing values. But I'm just assuming now to convert to bool it needs to have chracters otherwise exception.

---
In reply to: [211317465](https://github.com/dotnet/machinelearning/pull/693#discussion_r211317465) [](ancestors = 211317465)